src/cloud_provider/aws/databases/mongodb.rs

```diff
@@ -92,13 +92,17 @@ impl MongoDB {
     }
 }
 
-impl StatefulService for MongoDB {
+impl StatefulService for MongoDbAws {
+    fn as_stateful_service(&self) -> &dyn StatefulService {
```

Comment:
OPT: maybe use `impl From<MongoDbAws> for StatefulService` ? I think you should add `From<MongoDbAws>` to `StatefulService` trait declaration.

Reply:
You can't because when you are in the From<xxxx> the compiler don't know that MongoDbAws is also a StatefulService
Rust does not support upcasting (yet) https://stackoverflow.com/questions/28632968/why-doesnt-rust-support-trait-object-upcasting